drivers/adc/adc_ads114s0x.c

```diff
@@ -0,0 +1,1025 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER 1
+#include "adc_context.h"
+
+LOG_MODULE_REGISTER(ads114s0x, CONFIG_ADC_LOG_LEVEL);
+
+#define ADS114S0X_CLK_FREQ_IN_MHZ			    4096
+#define ADS114S0X_RESET_LOW_TIME_IN_CLOCK_CYCLES	    4
+#define ADS114S0X_START_SYNC_PULSE_DURATION_IN_CLOCK_CYCLES 4
+#define ADS114S0X_SETUP_TIME_IN_CLOCK_CYCLES		    32
+#define ADS114S0X_INPUT_SELECTION_AINCOM		    12
+#define ADS114S0X_RESOLUTION				    16
+#define ADS114S0X_REF_INTERNAL				    2500
+
+/*
+ * According to the datasheet this should be 4096 * 1/fclk = 1us,
+ * but it actually takes more time for the ADC to report ready in
+ * its status register.
+ */
+#define ADS114S0X_RESET_DELAY_TIME_IN_US 1000
+
+#define ADS114S0X_RESET_LOW_TIME_IN_US                                                             \
+	(ADS114S0X_RESET_LOW_TIME_IN_CLOCK_CYCLES / ADS114S0X_CLK_FREQ_IN_MHZ)
+#define ADS114S0X_START_SYNC_PULSE_DURATION_IN_US                                                  \
+	(ADS114S0X_START_SYNC_PULSE_DURATION_IN_CLOCK_CYCLES / ADS114S0X_CLK_FREQ_IN_MHZ)
+#define ADS114S0X_SETUP_TIME_IN_US                                                                 \
+	(ADS114S0X_SETUP_TIME_IN_CLOCK_CYCLES / ADS114S0X_CLK_FREQ_IN_MHZ)
```

Comment:
These time values will currently get the value of 0 because of the incorrect frequency used.

Reply:
done